dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -25,691 +16,357 @@
-     * @throws DotSecurityException 
-     *            
+     * @throws DotDataException
+     * 
      */
-    private static void refreshPageInCache(String pageIdentifier) throws DotDataException, DotSecurityException {
+    private static void refreshPageInCache(String pageIdentifier) throws DotDataException {
         Set<String> inodes = new HashSet<String>();
-        List<ContentletVersionInfo> infos = APILocator.getVersionableAPI().findContentletVersionInfos(pageIdentifier);
+        List<ContentletVersionInfo> infos = APILocator.getVersionableAPI()
+            .findContentletVersionInfos(pageIdentifier);
         for (ContentletVersionInfo versionInfo : infos) {
             inodes.add(versionInfo.getWorkingInode());
-            if(versionInfo.getLiveInode() != null){
-              inodes.add(versionInfo.getLiveInode());
+            if (versionInfo.getLiveInode() != null) {
+                inodes.add(versionInfo.getLiveInode());
             }
         }
+        try {
+            List<Contentlet> contentlets = APILocator.getContentletAPIImpl()
+                .findContentlets(Lists.newArrayList(inodes));
+            for (Contentlet pageContent : contentlets) {
+                IHTMLPage htmlPage = APILocator.getHTMLPageAssetAPI()
+                    .fromContentlet(pageContent);
+                PageServices.invalidateAll(htmlPage);
+            }
+        } catch (DotStateException | DotSecurityException e) {
+            Logger.warn(MultiTreeFactory.class, "unable to refresh page cache:" + e.getMessage());
+        }
+    }
+
+
+    private static MultiTree dbToMultiTree(Map<String, Object> row) {
+        final String relationType = (String) row.getOrDefault("relation_type", null);
+        final String parent1 = (String) row.getOrDefault("parent1", null);
+        final String parent2 = (String) row.getOrDefault("parent2", null);
+        final String child = (String) row.getOrDefault("child", null);
+        final int order = Integer.valueOf((Integer) row.getOrDefault("tree_order", 0));
+        return new MultiTree(parent1, parent2, child, relationType, order);
+
+
+    }
 
-        List<Contentlet> contentlets = APILocator.getContentletAPIImpl().findContentlets(Lists.newArrayList(inodes));
-		for (Contentlet pageContent : contentlets) {
-			IHTMLPage htmlPage = APILocator.getHTMLPageAssetAPI().fromContentlet(pageContent);
-			PageServices.invalidateAll(htmlPage);
-		}
+    public static List<MultiTree> dbToMultiTree(List<Map<String, Object>> dbRows) {
+        return (List<MultiTree>) dbRows.stream()
+            .map(row -> dbToMultiTree(row))
+            .collect(Collectors.toList());
     }
-	
+
+    /**
+     * {link {@link #saveMultiTree(MultiTree)} The multitree does not respect language
+     * 
+     * @param multiTreeEN
+     * @param english
+     * @throws DotDataException
+     */
+    @Deprecated
+    public static void saveMultiTree(MultiTree multiTree, long lang) throws DotDataException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)